permits/forms.py

```diff
@@ -692,11 +692,20 @@ def __init__(self, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
 
-        self.fields["geom"].widget.attrs["options"] = self.get_widget_options(
-            self.permit_request
-        )
+        if not services.geotime_needs_info(self.permit_request, "date"):
```

Comment:
Je changerais 2 trucs ici:

* Utiliser des chaînes de caractères comme "types d’information" peut causer des bugs si tu as une typo dans une chaîne, et ça rend le code plus dur à lire parce qu’on ne peut pas savoir quelles sont les valeurs possibles. Au lieu des chaînes `"date"` et `"geometry"` j’utiliserais plutôt [un enum.](https://docs.python.org/3/library/enum.html)
* Chaque exécution de `geotime_needs_info` va exécuter au moins 2 requêtes SQL. Ce serait plus efficace d’avoir une méthode `get_geotime_required_info` qui renvoie un `set` de champs requis, et tu peux ensuite tester ici si `GeoTimeInfo.DATE in required_info`.

Reply:
done